iotdb-api/udf-api/src/main/java/org/apache/iotdb/udf/api/UDTF.java

```diff
@@ -112,6 +116,48 @@ default void transform(RowWindow rowWindow, PointCollector collector) throws Exc
     // do nothing
   }
 
+  /**
+   * When the user specifies {@link MappableRowByRowAccessStrategy} to access the original data in
+   * {@link UDTFConfigurations}, this method will be called to process the transformation. Compared
+   * to {@link #transform(Row)}, this method processes input data in batches. In a single UDF query,
+   * this method may be called multiple times.
+   *
+   * @param columns original input data columns (aligned by time)
+   * @param builder used to collect output data points
+   * @throws Exception the user can throw errors if necessary
+   * @throws UnsupportedOperationException if the user does not override this method
+   * @see MappableRowByRowAccessStrategy
+   */
+  default void transform(Column[] columns, ColumnBuilder builder) throws Exception {
+    int colCount = columns.length;
+    int rowCount = columns[0].getPositionCount();
+
+    // collect input data types from columns
+    TSDataType[] dataTypes = new TSDataType[colCount];
+    for (int i = 0; i < colCount; i++) {
+      dataTypes[i] = columns[i].getDataType();
+    }
```

Comment:
In my view, this interface, like other interfaces, should have a default empty implementation, allowing users to implement the specific interface. So I'm curious about the purpose of this implementation and RowImpl.

Reply:
I see👍